client/components/tinymce/plugins/mentions/suggestion-list.jsx

```diff
@@ -0,0 +1,90 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+
+/**
+ * Internal dependencies
+ */
+import PopoverMenu from 'components/popover/menu';
+import PopoverMenuItem from 'components/popover/menu-item';
+import Suggestion from './suggestion';
+
+const SuggestionList = React.createClass( {
+	displayName: 'SuggestionList',
+
+	propTypes: {
+		query: React.PropTypes.string,
+		suggestions: React.PropTypes.array,
+		isVisible: React.PropTypes.bool,
+		onClick: React.PropTypes.func,
+		onClose: React.PropTypes.func
+	},
+
+	getDefaultProps() {
+		return {
+			query: '',
+			suggestions: [],
+			isVisible: false
+		};
+	},
+
+	getMatchingSuggestions() {
+		const { suggestions, query } = this.props;
+		const matcher = new RegExp( "^" + query + "| " + query, 'ig' ); // Start of string or preceded by a space.
+		let matchingSuggestions = suggestions;
+
+		if ( query.length > 0 ) {
+			matchingSuggestions = [];
+
+			for ( let i = 0, len = suggestions.length; i < len; i++ ) {
```

Comment:
This looks like a good candidate for [`Array#filter`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/filter):

```js
if ( query.length > 0 ) {
	matchingSuggestions = matchingSuggestions.filter( ( suggestion ) => {
		const name = suggestion.name || suggestion.user_login + ' ' + suggestion.display_name;

		return name.toLowerCase().match( matcher );
	} );
}
```